src/op_analytics/cli/subcommands/pulls/agora/delegate_events.py

```diff
@@ -0,0 +1,170 @@
+from dataclasses import dataclass
+from op_analytics.cli.subcommands.pulls.agora.data_access import (
+    Agora,
+    write,
+    _camelcase_to_snakecase,
+    parse_isoformat_with_z,
+)
+from op_analytics.coreutils.logger import structlog
+from op_analytics.coreutils.request import get_data
+from op_analytics.coreutils.threads import run_concurrently
+
+# from op_analytics.coreutils.time import parse_isoformat_with_tz
+from typing import Any
+import requests
+from tenacity import retry, stop_after_attempt, wait_exponential, retry_if_exception_type
```

Comment:
lets use "stamina" instead of "tenacity".  We are already using stamina on other parts so it would be good to make that uniform.   

That said I think we should try to rely only on the two types of retries that are supported on request.py.  We have a Retry adapter and then also a retrier that retries on all exceptions just in case (we use this for defillama where the kind of error we may get is not always clear).

Reply:
Reinvented the wheel here, removing all extra logic and using the retry logic provided by requests.py